### PR TITLE
Update docs for run_options

### LINKED ISSUE #240: An user should be able to verify that their run options were successfully used
**What is the expected feature or enhancement?**
It is impossible to get verification that `run_options` other than `shots` are being applied successfully.  There should be some way to verify that the program did indeed use the options I send it.

**Acceptance criteria**
The user has some notification, in the return object or the like, that the requested options were applied.

### PR DESCRIPTION
**Pull request by kt474**
### Summary

Current list of acceptable `run_options`:

```
shots 
qubit_lo_freq
meas_lo_freq
schedule_los
rep_delay
init_qubits
use_measure_esp
```
There is currently no way to actually confirm that `run_options` are being applied but these are all the options in the `assemble()` method that apply to the primitives. This PR only updates the relevant docstrings 
### Details and comments
Fixes #240

Related to #243, #302

### CODE REVIEW

**Review by rathishcholarajan — COMMENTED** [review]
@kt474 Have you tried all these options and do all these work?

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
This needs to be tested with #240 before merging.

**Comment by kt474** [comment]
@rathishcholarajan @renier @kwoane thoughts on updating these docstrings?

I'm also ok with just closing this PR because we are updating the docs [here](https://quantum-computing.ibm.com/services/programs?program=sampler) in ntc-programs PR #210

File: qiskit_ibm_runtime/qiskit/primitives/base_estimator.py [hunk 1]
```diff
@@ -278,7 +278,9 @@ def __call__(
             circuits: the list of circuit indices or circuit objects.
             observables: the list of observable indices or observable objects.
             parameter_values: concrete parameters to be bound.
-            run_options: runtime options used for circuit execution.
+            run_options: runtime options used for circuit execution. These include shots,
+                qubit_lo_freq, meas_lo_freq, schedule_los, rep_delay, init_qubits,
+                and use_measure_esp.
```
Comment (rathishcholarajan) [inline qiskit_ibm_runtime/qiskit/primitives/base_estimator.py:283]: These base classes are copied from terra, let's not update these. just sampler.py and estimator.py should be good enough for this PR.

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]

**Comment by renier** [comment]
> @rathishcholarajan @renier @kwoane thoughts on updating these docstrings?
> 
> I'm also ok with just closing this PR because we are updating the docs [here](https://quantum-computing.ibm.com/services/programs?program=sampler) in ntc-programs PR # 210

I'm fine with updating the docstrings. I just would like to verify that these run_options are actually leveraged in the runtime environment. @rathishcholarajan will probably need your help. This verification can happen as its own issue, though. It doesn't have to block the doc. We can come back and add/update/remove once we investigate run_options more.

**Comment by renier** [comment]
Let's go ahead without the verification. All the run options are being passed down. Whether they are honored depends on the backend capabilities.

**Review by rathishcholarajan — APPROVED** [review]

**Pull request merged**